### PR TITLE
Update Orleans quickstart with AZD template

### PR DESCRIPTION
**Pull request by seesharprun**
## Summary

Update the Orleans "Deploy & Scale" quickstart to use an AZD template ([`orleans-url-shortener`](https://github.com/azure-samples/orleans-url-shortener))

## Key motivations

- Reduces the complexity of the quickstart. Now, you simply init the template, deploy the in-memory sample as a container, add packages, change code, and then redeploy the container.
- Reduces screenshots. Since AZD deploys the application architecture, there's no need to maintain screenshots that quickly age.
- Simplifies changes. The AZD template is already dockerized, so the only changes the user needs to make are to import the package and implement new logic for the silo builder.
- Shows best practices. The new template doesn't hard-code strings. It uses RBAC and the system-assigned managed identity of the container app. The deploying user is also given RBAC permissions to make local debugging easier.
- Showcasing the quickstart. The AZD template can be added to various platforms like <https://github.com/azure/awesome-azd> and reference back to the quickstart increasing visibility.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md](https://github.com/dotnet/docs/blob/31ef0614cedc361eefc528aca5951d3eed485aaf/docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md) | [CustomerIntent: As a developer, I want to host my Orleans application in Azure so that I can take advantage of the scaling capabilities for the database and application services.](https://review.learn.microsoft.com/en-us/dotnet/orleans/quickstarts/deploy-scale-orleans-on-azure?branch=pr-en-us-39099) |

### CODE REVIEW

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 1]
```diff
@@ -1,96 +1,232 @@
...
-ms.date: 10/03/2023
+ms.date: 01/11/2024
 zone_pivot_groups: orleans-persistence-option
 # CustomerIntent: As a developer, I want to host my Orleans application in Azure so that I can take advantage of the scaling capabilities for the database and application services.
 ---
 
 # Deploy and scale an Orleans app on Azure
 
-In this quickstart, you deploy and scale an Orleans URL shortener app on Azure Container Apps. The app allows users to submit a full URL to the app, which returns a shortened version they can share with others to direct them to the original site. Orleans and Azure provide the scalability features necessary to host high traffic apps like URL shorteners. Orleans is also compatible with any other hosting service that supports .NET.
+In this quickstart, you deploy and scale an Orleans URL shortener app on Azure Container Apps. The app allows users to submit a full URL to the app, which returns a shortened version they can share with others to direct them to the original site.  Orleans and Azure provide the scalability features necessary to host high traffic apps like URL shorteners. Orleans is also compatible with any other hosting service that supports .NET.
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:13]:
```suggestion
In this quickstart, you deploy and scale an Orleans URL shortener app on Azure Container Apps. The app allows users to submit a full URL to the app, which returns a shortened version they can share with others to direct them to the original site. Orleans and Azure provide the scalability features necessary to host high traffic apps like URL shorteners. Orleans is also compatible with any other hosting service that supports .NET.
```

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 2]
```diff
@@ -1,96 +1,232 @@
...
+1. Authenticate to the Azure Developer CLI using `azd auth login`. Follow the steps specified by the tool to authenticate to the CLI using your preferred Azure credentials.
 
-Prior to using the grain, you must install the corresponding **clustering** and **persistence** NuGet packages.
+    ```azurecli
+    azd auth login
+    ```
+
+1. Get the sample application using the AZD template `orleans-url-shortener` and the `azd init` command.
+
+    ```azure-cli
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:42]: `azd` is the `azuredeveloper` slug.

```suggestion
    ```azuredeveloper
```

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 3]
```diff
@@ -1,96 +1,232 @@
...
-```bash
-git clone https://github.com/Azure-Samples/build-your-first-orleans-app-aspnetcore
-```
+1. Open a terminal in an empty directory.
 
-## Install the NuGet packages
+1. Authenticate to the Azure Developer CLI using `azd auth login`. Follow the steps specified by the tool to authenticate to the CLI using your preferred Azure credentials.
 
-Prior to using the grain, you must install the corresponding **clustering** and **persistence** NuGet packages.
+    ```azurecli
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:36]:
```suggestion
    ```azuredeveloper
```

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 4]
```diff
@@ -1,96 +1,232 @@
...
+    ```
+
+1. During initialization, configure a unique environment name.
+
+    > [!TIP]
+    > The environment name will also be used as the target resource group name. For this quickstart, consider using `msdocs-orleans-url-shortener`.
+
+1. Deploy the Azure Cosmos DB for NoSQL account using `azd up`. The Bicep templates also deploy a sample web application.
+
+    ```azurecli
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:53]:
```suggestion
    ```azuredeveloper
```

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 5]
```diff
@@ -1,96 +1,232 @@
...
 
 ::: zone pivot="azure-storage"
 
-| | NuGet package |
-| --- | --- |
-| **Clustering** | [Microsoft.Orleans.Clustering.AzureStorage](https://www.nuget.org/packages/Microsoft.Orleans.Clustering.AzureStorage) |
-| **Persistence** | [Microsoft.Orleans.Persistence.AzureStorage](https://www.nuget.org/packages/Microsoft.Orleans.Persistence.AzureStorage) |
+1. Using the terminal, run `azd env set` to configure the `DEPLOY_AZURE_COSMOS_DB_NOSQL` environment variable to enable deployment of Azure Cosmos DB for NoSQL.
+
+    ```azurecli
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:93]:
```suggestion
    ```azuredeveloper
```

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 6]
```diff
@@ -1,96 +1,232 @@
...
 
 ::: zone pivot="azure-cosmos-db-nosql"
 
-| | NuGet package |
-| --- | --- |
-| **Clustering** | [Microsoft.Orleans.Clustering.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Clustering.Cosmos) |
-| **Persistence** |  [Microsoft.Orleans.Persistence.Cosmos](https://www.nuget.org/packages/Microsoft.Orleans.Persistence.Cosmos) |
+1. Using the terminal, run `azd env set` to configure the `DEPLOY_AZURE_COSMOS_DB_NOSQL` environment variable to enable deployment of Azure Cosmos DB for NoSQL.
+
+    ```azurecli
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:103]:
```suggestion
    ```azuredeveloper
```

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 7]
```diff
@@ -1,96 +1,232 @@
...
+    azd env set DEPLOY_AZURE_TABLE_STORAGE true
+    ```
 
 ::: zone-end
 
-## Configure the sample app
+1. Run `azd provision` to redeploy your application architecture with the new configuration. Wait for the provisioning process to complete. The process can take **approximately two minutes**.
 
-The sample app is currently configured to create a localhost cluster and persist grains in memory. When hosted in Azure, Orleans can be configured to use more scalable, centralized state using services like Azure Table Storage and Azure Blob Storage.
+    ```azurecli
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:111]:
```suggestion
    ```azuredeveloper
```

File: docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md [hunk 8]
```diff
@@ -99,213 +235,152 @@ Update the `builder` configuration code in the _Program.cs_ file to match the ex
...
-    * **Resource group**: Select **Create new** and then enter a name of *msdocs-url-shortener*.
-    * **Storage account name**: Enter the name *OrleansUrlShortenerXXX* where XXX are unique numbers. Storage account names must be unique across Azure.
-    * **Region**: Select a region that is near your location.
-    * Leave the rest of the options at their defaults, and then select **Review**.
-    * Select **Create** after Azure validates your settings.
-
-        :::image type="content" source="../media/create-storage-account.png" alt-text="A screenshot showing how to create a storage account." lightbox="../media/create-storage-account.png":::
-
-1. Select **Go to resource** after the storage account is created.
+    ```azurecli
```
Comment (IEvangelist) [inline docs/orleans/quickstarts/deploy-scale-orleans-on-azure.md:286]:
```suggestion
    ```azuredeveloper
```

**Review by IEvangelist — APPROVED** [review]
LGTM, a few minor nits.

**Comment by seesharprun** [comment]
@IEvangelist, you beat me to the changes 😆

**Pull request merged**